pkgs/os-specific/linux/busybox/default.nix

```diff
@@ -75,7 +75,7 @@ stdenv.mkDerivation rec {
       url = "https://git.alpinelinux.org/aports/plain/main/busybox/0002-nslookup-sanitize-all-printed-strings-with-printable.patch?id=ed92963eb55bbc8d938097b9ccb3e221a94653f4";
       sha256 = "sha256-vl1wPbsHtXY9naajjnTicQ7Uj3N+EQ8pRNnrdsiow+w=";
     })
-  ] ++ lib.optional (stdenv.hostPlatform != stdenv.buildPlatform) ./clang-cross.patch;
+  ] ++ lib.optional (!stdenv.cc.isGNU) ./replace-gcc-with-cc.patch;
```

Comment:
While this fixes the issue in your case, would it break cross-compilation with GCC? `!stdenv.cc.isGNU` is likely not a superset of `stdenv.hostPlatform != stdenv.buildPlatform`.

Reply:
That's a good observation.
Perhaps I should not worry about triggering a rebuild and always apply the patch for the sake of consistency?

Nevertheless, cross-compilation still works (tested with `pkgsCross.riscv64.busybox`).

Reply:
It's always good practice to apply patches unconditionally when there's no negative effect for the systems it doesn't do anything on, because otherwise it's easy to miss that patches need to be updated for cases not being tested.  (Rebuilds don't count as a negative effect except in time sensitive situations.  It's a rebuild once vs. a more complicated package forever.)